### PR TITLE
Add macOS files to .gitignore

### PR DESCRIPTION
**Pull request by RDIL**
## PR Summary

Apple files have no place here at Microsoft

## Label   
I would suggest CL-Tools.  

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [X] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed - Issue link:
- **Testing - New and feature**
    - [X] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by RDIL** [comment]
@iSazonov @PaulHigin @SteveL-MSFT requesting review

**Review by SteveL-MSFT — COMMENTED** [review]
Looks ok to me, but I've never generated any of these files that they would accidentally be included in a commit

**Comment by RDIL** [comment]
They are all pretty common for avid Mac users

File: .gitignore [hunk 1]
```diff
@@ -60,6 +60,20 @@ gen
 
 # macOS
 .DS_Store
+.DocumentRevisions-V100
+.fseventsd
+.Spotlight-V100
+.TemporaryItems
+.Trashes
+.VolumeIcon.icns
+.com.apple.timemachine.donotpresent
```
Comment (TravisEz13) [inline .gitignore]: some of these files should only be present at the root of the drive, but I'm not complaining unless they cause issues.

**Review by TravisEz13 — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (RDIL) [reply .gitignore]: Hah, funny you should mention that, I have my PowerShell install on the root drive!!

Comment (TravisEz13) [reply .gitignore]: Just a couple of them I had to lookup.

**Pull request merged**